provider/ansible.py

```diff
@@ -167,16 +172,33 @@ def distro_install(packages="ansible"):
     policy_map = {"distro_install": distro_install,
                   "python_install": python_install}
 
+    ansible_install_policy = params.get('ansible_install_policy')
+    if ansible_install_policy:
+        if ansible_install_policy not in policy_map:
+            LOG_JOB.error(f"No valid install policy: {ansible_install_policy}.")
+            return False
     package_list = params.get_list("package_list", 'sshpass')
     try:
-        path.find_command('ansible-playbook')
-    except path.CmdNotFoundError:
-        ansible_install_policy = params['ansible_install_policy']
-        if ansible_install_policy == 'distro_install':
+        check_cmd = params.get("ansible_check_cmd")
+        if ansible_install_policy == 'python_install':
+            pip_bin = _pip_binary()
+            if not pip_bin:
+                return False
+            check_cmd = rf"{pip_bin} freeze | grep -v ansible-core | grep -q ansible="
```

Comment:
Logically, it will overwrite the setting in the conf  in L182,  is this the expect behaviour ?

Reply:
Yes, I can confirm that :+1:  `check_cmd` is intended to be customizable for the case of `distro_install` (because we don't know about the distro used to run the test), but it is fixed for `python_install` as pip always behaves the same.